src/game/client/draw/w3dtruckdraw.cpp

```diff
@@ -0,0 +1,631 @@
+        { "MidRightMidTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTruckDrawModuleData, m_midMidRightTireBoneName) },
+        { "TireRotationMultiplier", &INI::Parse_Real, nullptr, offsetof(W3DTruckDrawModuleData, m_rotationSpeedMultiplier) },
+        { "PowerslideRotationAddition", &INI::Parse_Real, nullptr, offsetof(W3DTruckDrawModuleData, m_powerslideRotationAddition) },
+        { "CabBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTruckDrawModuleData, m_cabBoneName) },
+        { "TrailerBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTruckDrawModuleData, m_trailerBoneName) },
+        { "CabRotationMultiplier", &INI::Parse_Real, nullptr, offsetof(W3DTruckDrawModuleData, m_cabRotationMultiplier) },
+        { "TrailerRotationMultiplier", &INI::Parse_Real, nullptr, offsetof(W3DTruckDrawModuleData, m_trailerRotationMultiplier) },
+        { "RotationDamping", &INI::Parse_Real, nullptr, offsetof(W3DTruckDrawModuleData, m_rotationDamping) },
+        { nullptr, nullptr, nullptr, 0 },
+    };
+    // clang-format on
+
+    W3DModelDrawModuleData::Build_Field_Parse(p);
+    p.Add(dataFieldParse, 0);
+}
+
+W3DTruckDraw::W3DTruckDraw(Thing *thing, ModuleData const *module_data) :
+    W3DModelDraw(thing, module_data),
+    m_effectsInitialized(false),
+    m_wasAirborne(false),
+    m_isPowersliding(false),
+    m_dustEffect(nullptr),
+    m_dirtEffect(nullptr),
+    m_powerslideEffect(nullptr),
+    m_frontWheelRotation(0.0f),
+    m_rearWheelRotation(0.0f),
+    m_midFrontWheelRotation(0.0f),
+    m_midRearWheelRotation(0.0f),
+    m_frontLeftTireBone(0),
+    m_frontRightTireBone(0),
+    m_rearLeftTireBone(0),
+    m_rearRightTireBone(0),
+    m_midFrontLeftTireBone(0),
+    m_midFrontRightTireBone(0),
+    m_midRearLeftTireBone(0),
+    m_midRearRightTireBone(0),
+    m_midMidLeftTireBone(0),
+    m_midMidRightTireBone(0),
+    m_prevRenderObj(nullptr)
+{
+    m_landingSound = *thing->Get_Template()->Get_Per_Unit_Sound("TruckLandingSound");
+    m_powerslideSound = *thing->Get_Template()->Get_Per_Unit_Sound("TruckPowerslideSound");
+}
+
+W3DTruckDraw::~W3DTruckDraw()
+{
+    Toss_Emitters();
+}
+
+NameKeyType W3DTruckDraw::Get_Module_Name_Key() const
+{
+    static const NameKeyType nk = g_theNameKeyGenerator->Name_To_Key("W3DTruckDraw");
+    return nk;
+}
+
+void W3DTruckDraw::Do_Draw_Module(const Matrix3D *transform)
+{
+    W3DModelDraw::Do_Draw_Module(transform);
+
+    if (g_theWriteableGlobalData->m_showClientPhysics) {
+        const W3DTruckDrawModuleData *data = Get_W3D_Truck_Draw_Module_Data();
+
+        if (data != nullptr) {
+            bool frozen = g_theTacticalView->Is_Time_Frozen() && !g_theTacticalView->Is_Camera_Movement_Finished();
+            frozen = frozen || g_theScriptEngine->Is_Time_Frozen_Debug() || g_theScriptEngine->Is_Time_Frozen_Script();
+
+            if (!frozen) {
+                Object *object = Get_Drawable()->Get_Object();

```

Comment:
`const Object *`

Reply:
Can confirm that this is calling the non-const version of Get_Object()